src/types/Firestore.ts

```diff
@@ -41,60 +41,54 @@ export interface FirestoreStatus {
 
 // note: these entries should be sorted alphabetically
 export interface FirestoreData {
```

Comment:
I'm guessing all of these changes have been tested and ensure that they aren't going to break anything? They seem unrelated to the specified purpose of this PR?

I personally would have done them as a separate PR to minimize the risk of the changes + simplify reviewing this PR.

Reply:
What kind of changes are you referring to? I suppose every value can be undefined here. I have just changed the types to be more accurate. I have made the types stricter, not looser, so I think this should be okay. They are unrelated, I agree.

Reply:
I guess so.. I probably would have preferred that it was done in isolation in https://github.com/sparkletown/sparkle/pull/744 though, as that would have reduced the effort of reviewing this PR + keeping everything more 'contained' to it's intended purpose.

Not worth undoing/etc here now of course. But in future, keep it in mind. Less is more. Smaller more 'constrained' PR's that try to tackle a single thing are generally better/easier than giant widely scoped PR's.

Reply:
Agreed